bitcoin/src/pow.rs

```diff
@@ -923,6 +935,110 @@ impl kani::Arbitrary for U256 {
     }
 }
 
+/// Error from `get_block_header_by_height` callback.
+///
+/// Occurse when there is an error while get a new block header
+/// from an external source.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum GetBlockHeaderError {
+    /// Block Header not found
+    NotFound,
+    /// Generic error with custom message
+    Generic(String),
+}
+
+impl fmt::Display for GetBlockHeaderError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let msg = match self {
+            Self::Generic(error) => error,
+            Self::NotFound => "Block header not found",
+        };
+        write!(f, "{}", msg)
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for GetBlockHeaderError {}
+
+/// Calculate the next Work required as the `GetNextWorkRequired` do
+/// in bitcoin core.
+///
+/// Return a error if the callback provide from the caller fails.
+#[cfg_attr(all(test, mutate), mutate)]
+pub fn calculate_next_work_required<F>(
```

Comment:
This function says `work` but actually returns target, those are not the same thing so either the name is wrong or the returned value. (I guess it's the name.)

Reply:
Its because the name is copied from Core, I agree, we should return a `CompactTarget` and call it `calculate_next_target` or some such name.

Reply:
Done! thanks